tls/s2n_x509_validator.c

```diff
@@ -549,3 +550,64 @@ s2n_cert_validation_code s2n_x509_validator_validate_cert_stapled_ocsp_response(
     return ret_val;
 #endif /* S2N_OCSP_STAPLING_SUPPORTED */
 }
+
+DEFINE_POINTER_CLEANUP_FUNC(STACK_OF(X509)*, wipe_cert_chain);
+
+s2n_cert_validation_code s2n_x509_validator_validate_certificate_signatures(struct s2n_x509_validator *validator,  struct s2n_connection *conn)
```

Comment:
Is it possible to make this function return S2N_RESULT, and the s2n_cert_validation_code can be an output variable?

Just a nitpick, but `struct s2n_connection` is typically the first argument to a function.

Reply:
I agree. It looks like you're only ever returning S2N_CERT_ERR_UNTRUSTED from this method anyway; explicitly on line 576, and implicitly when you call S2N_ERROR_IF and GUARD_AS_POSIX.

Reply:
Yeah, the other validation functions have this particular return type, which is s2n_cert_validation, but I don't really need it in this function.